lib/kitchen/document.rb

```diff
@@ -177,6 +177,18 @@ def modified_id_to_paste(original_id)
       end
     end
 
+    # Copies an ID, returning a unique ID
+    #
+    # @param id [String]
+    #
+    def copy_id(id)
+      return nil if id.nil?
+      return '' if id.blank?
+
+      @next_paste_count_for_id[id] ? @next_paste_count_for_id[id] += 1 : @next_paste_count_for_id[id] = 1
+      "#{id}#{@id_copy_suffix}#{@next_paste_count_for_id[id]}"
+    end
+
```

Comment:
To me, this exposes some of the ID incrementing functionality that had previously been nicely encapsulated; however, it is useful in the two modified bake directions. Thoughts?

Reply:
This method may not be necessary, but if it is let's revisit these questions.